examples/CSharp/OpenGL Tutorials/Tutorial 1.3 - Textures/Program.cs

```diff
@@ -0,0 +1,306 @@
+                // 5. The image's height.
+                // 6. This is the image's border. This valu MUST be 0. It is a leftover component from legacy OpenGL, and
+                //    it serves no purpose.
+                // 7. Our image data is formatted as RGBA data, therefore we must tell OpenGL we are uploading RGBA data.
+                // 8. StbImageSharp returns this data as a byte[] array, therefore we must tell OpenGL we are uploading
+                //    data in the unsigned byte format.
+                // 9. The actual pointer to our data!
+                _gl.TexImage2D(TextureTarget.Texture2D, 0, InternalFormat.Rgba, (uint) result.Width, 
+                    (uint) result.Height, 0, PixelFormat.Rgba, PixelType.UnsignedByte, ptr);
+            }
+
+            // Let's set some texture parameters!
+            // This tells the GPU how it should sample the texture.
+            
+            // Set the texture wrap mode to repeat.
+            // The texture wrap mode defines what should happen when the texture coordinates go outside of the 0-1 range.
+            // In this case, we set it to repeat. The texture will just repeatedly tile over and over again.
+            // You'll notice we're using S and T wrapping here. This is OpenGL's version of the standard UV mapping you
+            // may be more used to, where S is on the X-axis, and T is on the Y-axis.
+            _gl.TextureParameter(_texture, TextureParameterName.TextureWrapS, (int) TextureWrapMode.Repeat);
+            _gl.TextureParameter(_texture, TextureParameterName.TextureWrapT, (int) TextureWrapMode.Repeat);
+
+            // The min and mag filters define how the texture should be sampled as it resized.
+            // The min, or minification filter, is used when the texture is reduced in size.
+            // The mag, or magnification filter, is used when the texture is increased in size.
+            // We're using bilinear filtering here, as it produces a generally nice result.
+            // You can also use nearest (point) filtering, or anisotropic filtering, which is only available on the min
+            // filter.
+            // You may notice that the min filter defines a "mipmap" filter as well. We'll go over mipmaps below.
+            _gl.TextureParameter(_texture, TextureParameterName.TextureMinFilter, (int) TextureMinFilter.LinearMipmapLinear);
+            _gl.TextureParameter(_texture, TextureParameterName.TextureMagFilter, (int) TextureMagFilter.Linear);
+
+            // Generate mipmaps for this texture.
+            // Note: We MUST do this or the texture will appear as black (this is an option you can change but this is
+            // out of scope for this tutorial).
+            // What is a mipmap?
+            // A mipmap is essentially a smaller version of the existing texture. When generating mipmaps, the texture
+            // size is continuously halved until it reaches 1x1, at which point mipmap generation stops.
```

Comment:
```
            // size is continuously halved until it reaches 1x1, at which point mipmap generation stops.
```
I don't think mipmaps go all the way to 1x1:
https://www.khronos.org/opengl/wiki/Texture#Mipmap_range

Reply:
I suggest something like this:
```cs
// A mipmap is essentially a smaller version of the existing texture. They are generated by halving the size
// of the texture over and over a certain amount of times.
```

The only issue I have is that this implies mipmaps are generated all the way down to 1x1